R/class-plotdata.R

```diff
@@ -49,25 +55,37 @@ newPlotdata <- function(.dt = data.table(),
                          ...,
                          class = character()) {
 
+  impute0cols <- c()
```

Comment:
not essential, but i feel like this makes me want a generic helper that can take a vector of our per-variable plot reference list things (another class we could use when we eventually refactor) and some predicate, and return the column names that are associated w the variables meeting the condition..

`impute0cols <- findColNamesByPredicate(c(xAxisVariable,yAxisVariable,...), function(x){x$naToZero == 'TRUE' })`

Reply:
might mean first writing/ finding a helper to identify which members of some vector meet the condition, similar to purrr::every except instead of a single T/F you get a vector of them. then wrap this more specific fxn around that more general one..

Reply:
or just do this maybe `unlist(lapply(list(xAxisVariable,yAxisVariable,zAxisVariable), function(x){if(x$naToZero == T) x$id}))`

up to you.